tools/stats/upload_test_stat_aggregates.py

```diff
@@ -0,0 +1,84 @@
+import argparse
+import ast
+import datetime
+import json
+import os
+import re
+from typing import Any, List, Union
+
+import rockset  # type: ignore[import]
+
+from tools.stats.upload_stats_lib import upload_to_s3
+
+
+def get_oncall_from_testfile(testfile: str) -> Union[List[str], None]:
+    path = f"test/{testfile}"
+    if not path.endswith(".py"):
+        path += ".py"
+    # get oncall on test file
+    try:
+        with open(path) as f:
+            for line in f:
+                if line.startswith("# Owner(s): "):
+                    possible_lists = re.findall("\[.*\]", line)
+                    if len(possible_lists) > 1:
+                        raise Exception("More than one list found")
+                    elif len(possible_lists) == 0:
+                        raise Exception("No oncalls found or file is badly formatted")
+                    oncalls = ast.literal_eval(possible_lists[0])
+                    return list(oncalls)
+    except Exception as e:
+        if "." in testfile:
```

Comment:
Just curious what this does?  Having a dot in the filename?  I guess you have an example at hand

Reply:
there's files like "functorh.some_random_test" that don't have an owner, but its pretty obvious it belongs to functorch. Its pretty common for functorch and distributed